### PR TITLE
update k8s to 1.25 and go to 1.19

### PR DESCRIPTION
**Pull request by everettraven**
**What this PR does / why we need it**:
Updates Kubernetes dependencies to 1.25 and Go to 1.19 to provide support for using `kubebuilder-declarative-pattern` with k8s 1.25

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Additional documentation**:

### CODE REVIEW

**Comment by everettraven** [comment]
@atoato88 this PR should now be in a good state to review. Please let me know if there is anything else needed, thanks!

File: doc.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ limitations under the License.
 // declarative Kubernetes operators to manage the lifecycle of a
 // deployment.
 //
-// Getting Started
+// # Getting Started
```
Comment (atoato88) [inline doc.go:21]: Is it paragraph markup on markdown format?

File: pkg/patterns/addon/doc.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ implements the addonsv1alpha1.CommonObject interface. The object represents the
 to deploy an instance of a specific Addon in the cluster. This pattern manages a
 Kubernetes deployment for the specific addon based on the Addon Object.
 
-Writing an Addon Operator
+# Writing an Addon Operator
```
Comment (atoato88) [inline pkg/patterns/addon/doc.go:29]: same comment above.

**Comment by atoato88** [comment]
Thank you for updating!
It looks good!
And CI tests have passed too.

**Review by atoato88 — COMMENTED** [review]
I left some comments about `doc.go`. 
I think it's something format update for documentation, but please let me know a background about this update.

File: doc.go (hunk 1 above)
Comment (everettraven) [reply doc.go:21]: This has to do with `gofmt` updates related to bumping the Go version to Go 1.19.

If you take a look at the **Headings** section of this documentation https://tip.golang.org/doc/comment you can see that as of Go 1.19 the new heading format is similar to Markdown and uses the `#` prefix:

> The # syntax was added in Go 1.19. Before Go 1.19, headings were identified implicitly by single-line paragraphs satisfying certain conditions, most notably the lack of any terminating punctuation.

> Gofmt reformats [lines treated as implicit headings](https://github.com/golang/proposal/blob/master/design/51082-godocfmt.md#headings) by earlier versions of Go to use # headings instead. If the reformatting is not appropriate—that is, if the line was not meant to be a heading—the easiest way to make it a paragraph is to introduce terminating punctuation such as a period or colon, or to break it into two lines. 

I hope this helps!

File: pkg/patterns/addon/doc.go (hunk 2 above)
Comment (everettraven) [reply pkg/patterns/addon/doc.go:29]: See: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/238#discussion_r964838580

File: doc.go (hunk 1 above)
Comment (atoato88) [reply doc.go:21]: Thank you for detailed and helpful comment.
I understood that. 👍

**Pull request merged**